components/tabs/src/schema.rs

```diff
@@ -7,27 +7,43 @@
 // syncChangeCounter/syncStatus nor a mirror etc.
 
 use rusqlite::{Connection, Transaction};
-use sql_support::open_database::{
-    ConnectionInitializer as MigrationLogic, Error as MigrationError, Result as MigrationResult,
+use sql_support::{
+    open_database::{
+        ConnectionInitializer as MigrationLogic, Error as MigrationError, Result as MigrationResult,
+    },
+    ConnExt,
 };
 
-// The payload is json and this module doesn't need to deserialize, so we just
-// store each "payload" as a row.
-// On each Sync we delete all local rows re-populate them with every record on
-// the server. When we read the DB, we also read every single record.
-// So we have no primary keys, no foreign keys, and really completely waste the
-// fact we are using sql.
+// The record is the TabsRecord struct in json and this module doesn't need to deserialize, so we just
+// store each client as its own row.
 const CREATE_SCHEMA_SQL: &str = "
     CREATE TABLE IF NOT EXISTS tabs (
-        payload TEXT NOT NULL
+        guid            TEXT NOT NULL PRIMARY KEY,
+        record          TEXT NOT NULL,
+        last_modified   INTEGER NOT NULL
     );
 ";
 
-pub struct TabsMigrationLogin;
+const CREATE_META_TABLE_SQL: &str = "
+    CREATE TABLE IF NOT EXISTS moz_meta (
+        key    TEXT PRIMARY KEY,
+        value  NOT NULL
+    )
+";
+
+pub(crate) static LAST_SYNC_META_KEY: &str = "last_sync_time";
+pub(crate) static GLOBAL_SYNCID_META_KEY: &str = "global_sync_id";
+pub(crate) static COLLECTION_SYNCID_META_KEY: &str = "tabs_sync_id";
+// Tabs stores this in the meta table due to a unique requirement that we only know
+// the list of connected clients when syncing, however getting list of tabs could be
```

Comment:
For clarity, to "the list of connected clients when syncing," maybe append something like "so we can translate from the tabs sync record ID to the FxA device id for the client" or something, as that's really the core reason we need to do this.